lib/addons.js

```diff
@@ -7,46 +9,50 @@ function Addons (heroku, postgres) {
+      addon = yield this.heroku.apps(app.name).addons().create({attachment: {name: attachment.name}, plan: addon.plan.name});
       console.log('done');
       return addon;
     } catch (err) {
-      console.error(`Skipped addon ${addon.plan.name}`);
-      if (err.body) {
-        console.error(` !    ${err.body.message}`);
-      } else {
-        console.error(err);
-      }
+      console.error(`Skipped addon ${cli.color.magenta(addon.plan.name)}`);
+      if (err.body) cli.error(err.body.message);
+      else          cli.error(err);
     }
   },
 
-  getAddons: function* (app) {
-    let addons = yield this.heroku.request({path: `/apps/${app.name}/addons`});
-    return _.partition(addons, function (addon) {
-      return addon.plan.name.startsWith('heroku-postgresql');
-    });
+  attachAddon: function* (app, addon, attachment) {
+    try {
+      process.stdout.write(`Attaching addon ${cli.color.magenta(attachment.name)} from ${cli.color.cyan(addon.app.name)} to ${cli.color.cyan(app.name)} as ${cli.color.green(attachment.name)}... `);
+      attachment = yield this.heroku.addonAttachments().create({addon: addon.id, app: app.id, name: attachment.name});
+      console.log('done');
+      return attachment;
+    } catch (err) {
+      console.error(`Skipped attachment ${cli.color.magenta(attachment.name)}`);
+      if (err.body) cli.error(err.body.message);
+      else          cli.error(err);
+    }
   },
 
   copyAddons: function* (oldApp, newApp, skipPG) {
-    let allAddons = yield this.getAddons(oldApp);
-    let pgdbs = allAddons[0];
-    let addons = allAddons[1];
-    if (addons.length > 0) {
-      for (let addon of addons) {
-        yield this.addAddon(newApp, addon);
-      }
-    }
-    if (!skipPG && pgdbs.length > 0) {
-      let fromConfig = yield this.heroku.apps(oldApp.name).configVars().info();
-      for (let db of pgdbs) {
-        let toDB = yield this.addAddon(newApp, db);
+    let addons = yield this.heroku.get(`/apps/${oldApp.name}/addons`);
+    let attachments = yield this.heroku.get(`/apps/${oldApp.name}/addon-attachments`);
+    for (let attachment of attachments) {
+      /*jshint -W083 */
+      let addon = addons.find(a => a.id === attachment.addon.id);
+      if (addon.plan.name.startsWith('heroku-postgresql')) {
+        if (skipPG) continue;
+        let fromConfig = yield this.heroku.apps(oldApp.name).configVars().info();
+        let toDB = yield this.addAddon(newApp, addon, attachment);
         if (toDB) {
           let toConfig = yield this.heroku.apps(newApp.name).configVars().info();
-          yield this.postgres.migrateDB(fromConfig, db, toConfig, toDB);
+          yield this.postgres.migrateDB(fromConfig, addon, toConfig, toDB);
         }
+      } else if (addon.app.name !== oldApp.name) {
+        yield this.attachAddon(newApp, addon, attachment);
+      } else {
+        yield this.addAddon(newApp, addon, attachment);
```

Comment:
If an add-on has two attachments between the same app and add-on, it looks like this will create two instances of the add-on instead (one for each original attachment). This holds true for the PG branch above, too. What we really want to happen is when we encounter a second attachment for an add-on that we've already forked, is to attach to that newly created add-on instead.